web_button_invisible/models/hide_button_rule.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 ooops404
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+
+from odoo import _, api, exceptions, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class HideButtonRule(models.Model):
+    _name = "hide.button.rule"
```

Comment:
`model.btn.rule` or something similar to leave the door open to more features rather than just hiding them